### ISSUE TITLE
ONNX Parsing fails: Network must have at least one output

### ISSUE THREAD

**Issue by gcp** [NVIDIA/TensorRT#286]
## Description

Convolutional Resnet-like exported from PyTorch 1.3 can't be parsed, fails with 'Network must have at least one output'. This issue has been reported a few times, usually against PyTorch 1.3, but note:

The model is exported like this:

    torch.onnx.export(net, net_input, "net.onnx",
                      input_names=['channels', 'masks'],
                      output_names=['policy', 'outcome'])

Inspecting the ONNX with https://lutzroeder.github.io/netron/ shows that the inputs and outputs are indeed properly marked in the ONNX. So this is really a bug of the TensorRT ONNX parser. Because there's multiple outputs at different layers having to manually hunt down the correct layer where the output is is very cumbersome (annoying to do automatically if network topology changes) - and it didn't use to be necessary.

## Environment

**TensorRT Version**:  7.0.0.11
**GPU Type**:  RTX 2080
**Nvidia Driver Version**:  441.22
**CUDA Version**:  10.2
**CUDNN Version**:  7.6.0.5
**Operating System + Version**:  Windows 10
**Python Version (if applicable)**:  3.6
**TensorFlow Version (if applicable)**: 
**PyTorch Version (if applicable)**:  1.3 stable
**Baremetal or Container (if container which image + tag)**:  bare

## Relevant Files

ONNX file can be sent to a developer, but I can't post it publicly.

This looks similar to issue #220, which claims it should've been fixed in TRT7. But it hasn't.

**Comment by RizhaoCai** [comment]
> This was actually a problem due to the ONNX parser no longer taking implicit batch dimensions - parsing fails, 0 layers are found, and this generates the "network must have at least one output" error.

So, any idea to solve it ?

**Comment by gcp** [comment]
Use an ONNX model with dynamic axes so you have an explicit batch dimension.

**Comment by rmccorm4** [comment]
I believe the ONNX model can have a fixed batch size (not dynamic), but yes, ONNX models have to be parsed with the `--explicitBatch` (trtexec) / `NetworkDefinitionCreationFlag.EXPLICIT_BATCH` (API) flag set

**Comment by RizhaoCai** [comment]
> > Use an ONNX model with dynamic axes so you have an explicit batch dimension.
> 
> I believe the ONNX model can also have a fixed batch size (not necessarily dynamic), but yes, the takeaway is that ONNX models have to be parsed with the `--explicitBatch` (trtexec) / `NetworkDefinitionCreationFlag.EXPLICIT_BATCH` (API) flag set in TensorRT 7

In my case, using PyTorch 1.2 can solve this problem.
